### PR TITLE
Reddit Rest Connector Added

### PR DESCRIPTION
**Pull request by Kanapriya**

### CODE REVIEW

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/assembly/assemble-connector.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/assembly/assemble-connector.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/assembly/assemble-connector.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<assembly>
+    <formats>
+        <format>zip</format>
+    </formats>
+
+    <includeBaseDirectory>false</includeBaseDirectory>
+
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/assembly/assemble-connector.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/component.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/component.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/AXIS2/config/test_axis2_server_9000.xml [hunk 4]
```diff
@@ -0,0 +1,379 @@
+<!--
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/AXIS2/config/test_axis2_server_9000.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/testng.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/testng.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/testng.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    
+    <test name="reddit-Connector-Test" preserve-order="true" verbose="2">
+        <packages>
+            <package name="org.wso2.carbon.connector.integration.test.reddit"/>
+        </packages>
+    </test>
+
+</suite>
+
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/testng.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/restRequest/reddit/getAccessToken.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "refreshToken":"42591910-ITXmTVhxqv5NUlE_FPQ_YiXxlQM",
+  "grantType":"refresh_token",
+  "clientId":"bCzdckhMAJ91sw",
+  "clientSecret":"3uBELYRHhZLPUznD3D8Tj1OpqwE"
+}
+
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/restRequest/reddit/getAccessToken.json]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/restRequest/reddit/esb_getAnyUserPublicMultisList_mandatory.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "apiUrlOauth":"%s(apiUrlOauth)",
+  "accessToken":"%s(accessToken)",
+  "userName":"%s(publicMultiUsername)",
+  "expandSrs":"%s(expandSrs)"
+}
+
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/restRequest/reddit/esb_getAnyUserPublicMultisList_mandatory.json]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/proxies/reddit/reddit.xml [hunk 9]
```diff
@@ -0,0 +1,296 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/test/resources/artifacts/ESB/config/proxies/reddit/reddit.xml]: Fixed in https://github.com/wso2-attic/esb-connectors/commit/c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getAnyUserPublicMultisList.xml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+<template name="getAnyUserPublicMultisList" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="userName" description=" Existing Reddit username"/>
+    <parameter name="expandSrs" description=" A boolean value"/>
+    <sequence>
+        <property name="uri.var.userName" expression="$func:userName"/>
+        <property name="uri.var.expandSrs" expression="$func:expandSrs"/>
+        <property name="host" value=""/>
+        <script language="js">
+            <![CDATA[
+        var apiUrlOauth = mc.getProperty('uri.var.apiUrlOauth');
```
Comment (keerthu) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getAnyUserPublicMultisList.xml]: Format the code.

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getCurrentUserInfo.xml [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+ ~ software distributed under the License is distributed on an
+ ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ ~ KIND, either express or implied.  See the License for the
+ ~ specific language governing permissions and limitations
+ ~ under the License.
+ -->
+<template name="getCurrentUserInfo" xmlns="http://ws.apache.org/ns/synapse">
+    <sequence>
+        <property name="host" value=""/>
+        <script language="js">
```
Comment (keerthu) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getCurrentUserInfo.xml]: Check formatting throughout all pr.

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/RedditAuthorization/getAccessTokenFromRefreshToken.xml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+        scope="transport"/>
+        <call>
+            <endpoint>
+                <http method="POST"
+                      uri-template="legacy-encoding:{uri.var.apiUrl}/api/{uri.var.apiVersion}/access_token?grant_type=refresh_token&amp;client_id={uri.var.clientId}&amp;client_secret={uri.var.clientSecret}&amp;refresh_token={uri.var.refreshToken}"/>
+            </endpoint>
+        </call>
+        <filter source="$axis2:HTTP_SC" regex="[2][0][0]">
+            <then>
+                <!--<property name="TRANSPORT_HEADERS" action="remove" scope="axis2"/>-->
```
Comment (keerthu) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/RedditAuthorization/getAccessTokenFromRefreshToken.xml]: Remove commented line.

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/pom.xml [hunk 13]
```diff
@@ -0,0 +1,329 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/pom.xml]: Fixed in c61da2ff6de0ced270ed9698ed4fbd8227a92f5b

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getAnyUserPublicMultisList.xml (hunk 10 above)
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getAnyUserPublicMultisList.xml]: Fixed in 97f634769126cd1b6cc8397e6a08dccab09fc313

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getCurrentUserInfo.xml (hunk 11 above)
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/IdentityAndRead/getCurrentUserInfo.xml]: Fixed in 97f634769126cd1b6cc8397e6a08dccab09fc313

File: reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/RedditAuthorization/getAccessTokenFromRefreshToken.xml (hunk 12 above)
Comment (Kanapriya) [inline reddit/reddit-connector/reddit-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/RedditAuthorization/getAccessTokenFromRefreshToken.xml]: Fixed in 97f634769126cd1b6cc8397e6a08dccab09fc313

**Pull request merged**